tests/mgbench/benchmark.py

```diff
@@ -362,9 +706,125 @@ def filter_benchmarks(generators, patterns):
                 log.success("Throughput: {:02f} QPS".format(ret["throughput"]))
 
                 # Save results.
-                results_key = [dataset.NAME, dataset.get_variant(), group, test, test_type]
+                results_key = [
+                    dataset.NAME,
+                    dataset.get_variant(),
+                    group,
+                    query,
+                    WITHOUT_FINE_GRAINED_AUTHORIZATION,
+                ]
                 results.set_value(*results_key, value=ret)
 
+        ## If there is need for authorization testing.
+        if args.with_authorization:
+            print("Running query with authorization")
+            vendor.start_benchmark()
+            client.execute(
+                queries=[
+                    ("CREATE USER user IDENTIFIED BY 'test';", {}),
+                    ("GRANT ALL PRIVILEGES TO user;", {}),
+                    ("GRANT CREATE_DELETE ON EDGE_TYPES * TO user;", {}),
+                    ("GRANT CREATE_DELETE ON LABELS * TO user;", {}),
+                ]
+            )
+            client = runners.Client(
+                args.client_binary,
+                args.temporary_directory,
+                args.bolt_port,
+                username="user",
+                password="test",
+            )
+            vendor.stop()
+
+            for group in sorted(queries.keys()):
+                for query, funcname in queries[group]:
+
+                    log.info(
+                        "Running query:",
+                        "{}/{}/{}/{}".format(
+                            group, query, funcname, WITH_FINE_GRAINED_AUTHORIZATION
+                        ),
+                    )
+                    func = getattr(dataset, funcname)
+
+                    query_statistics = tail_latency(vendor, client, func)
+
+                    # Get number of queries to execute.
+                    # TODO: implement minimum number of queries, `max(10, num_workers)`
```

Comment:
Feel free to implement 😄

Reply:
I need to balance this out bcs of that long-running expansion test, I will do this on the fly, today/tomorrow.